datastorage.rb

```diff
@@ -0,0 +1,97 @@
+require 'json'
+require_relative 'person'
+require_relative 'options'
+require_relative 'rental'
+require_relative 'book'
+
+module DataStorage
+  def save_data(filename, data)
+    File.open(filename, 'w') do |file|
+      file.puts data.to_json
+    end
+  end
+
+  def load_data(filename)
+    JSON.parse(File.read(filename))
+  end
+
+  def save_person
+    data = []
+    @people.each do |person|
+      data << if person.instance_of?(Teacher)
+                { key: 'teacher', id: person.id, age: person.age, name: person.name,
+                  specialization: person.specialization }
+              else
+                { key: 'student', id: person.id, age: person.age, name: person.name,
+                  parent_permission: person.parent_permission, classroom: person.classroom }
+              end
+      save_data('person.json', data)
+    end
+  end
```

Comment:
- [ ] if the user opens the application and adds data to all the columns, then closes it and tries to open the application again they receive below error.
![Screenshot 2022-02-09 120956](https://user-images.githubusercontent.com/39240820/153187453-33a80918-88a9-468d-8587-995d77b07d88.png)
when looking at the .json file of person after that the name and age are being switched around, it seems that is the problem.
![Screenshot 2022-02-09 121436](https://user-images.githubusercontent.com/39240820/153188583-4e5e819d-67b9-4f39-a570-c8f60840848e.png)